### PR TITLE
Initial Service Manager specification

### PR DESCRIPTION
**Pull request by dpanayotov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,89 @@
...
+The main part is the core component.
+It is the central service broker registry that tracks all service instances and allows service broker and platform registration.
+This core component communicates with the registered brokers and acts as a platform per OSB for them.
+
+In each Platform Instance resides a component called the Service Broker Proxy.
+It is the substitute for all brokers registered at the Service Manager
+in order to replicate broker registration and access visibility changes in the corresponding Platform Instance.
+
+When a broker is registered or deregistered at the Service Manager,
+the Service Broker Proxy registers respectively deregisters itself in the name of this service broker.
```
Comment (NickyMateev) [inline README.md]: Maybe a more proper wording: "_the Service Broker Proxy respectively registers or deregisters itself in..._"

File: api.md [hunk 2]
```diff
@@ -0,0 +1,887 @@
...
+
+Each platform must know which service brokers need to be registered in it. It is the job of the service broker proxy to request the service brokers that need to be registered in its platform.
+
+**Note:** This is experimental API and should not be used in productive environment.
+
+### Request
+
+#### Route
+
+`GET /v1/internal/service_brokers`
```
Comment (KirilKabakchiev) [inline api.md]: Remove this api from the spec - the concept around the communication between the proxies and the service manager is not yet finalized. We should include information about it when it is done.

File: api.md [hunk 3]
```diff
@@ -0,0 +1,887 @@
...
+
+## Overview
+
+The Service Manager API defines an HTTP interface that allows the management of platforms, brokers and services from a central place. In general, the Service Manager API can be split into two groups - a Service Controller API that allows the management of platforms and service brokers and an OSB compliant API. The latter implements the [Open Service Broker (OSB) API](https://github.com/openservicebrokerapi/servicebroker/) and allows the Service Manager to act as a broker.
+
+One of the access channels to the Service Manager is via a CLI. The API should play nice in this context.
+A CLI-friendly string is all lowercase, with no spaces. Keep it short -- imagine a user having to type it as an argument for a longer command.
+
+## Platform Management
+
```
Comment (dzahariev) [inline api.md]: There is no GET ALL for the platforms. So there is not a way to get how much platforms are registered, and if you want to check if some platform is already registered you need to add it and to check the response - expecting 409 if was present and 200 if it is present from now on.

File: api.md [hunk 4]
```diff
@@ -0,0 +1,887 @@
...
+#### Body
+
+The response body MUST be a valid JSON Object (`{}`).
+
+```json
+{
+    "id": "36931aaf-62a7-4019-a708-0e9abf7e7a8f",
+    "name": "service-broker-name",
+    "description": "Service broker providing some valuable services",
+    "credentials": {
```
Comment (dzahariev) [inline api.md]: Why we return the whole request body as response?
Probably is sufficient to return the ID and a name, but anyway - the credentials should be excluded for sure.

File: api.md [hunk 5]
```diff
@@ -0,0 +1,887 @@
...
+        "basic": {
+            "username": "admin",
+            "password": "secret"
+        }
+    },
+    "created_at": "2016-06-08T16:41:26Z",
```
Comment (dzahariev) [inline api.md]: created_at and updated_at probably shall be different when the broker is updated.

File: README.md [hunk 6]
```diff
@@ -1,5 +1,89 @@
...
+This specification uses the terminology of the [Open Service Broker (OSB) API](https://github.com/openservicebrokerapi/servicebroker/).
+
+Additionally, this specification defines the following terms:
+
+- *Platform Type*: A concrete implementation of a platform. Examples are Cloud Foundry and Kubernetes.
+
+- *Platform Instance*: A deployment of a platform. For example, a Kubernetes cluster.
+
+- *Cloud Landscape*: A collection of Platform Instances of the same or different types.
+  In many cases, a Cloud Landscape is operated by one cloud provider,
```
[the remark this answers is not in the document]
Comment (dpanayotov) [reply README.md]: Clouds are operated by cloud providers, be it SAP, IBM, Google, etc and not by the IaaS provider

File: api.md [hunk 7]
```diff
@@ -0,0 +1,887 @@
+
+# Service Manager API
```
[the remark this answers is not in the document]
Comment (dpanayotov) [reply api.md]: Won't be only Control API. SM API = OSB + Control + Aggregated Catalog

File: api.md (hunk 3 above)
Comment (dpanayotov) [reply api.md]: Will be added in another PR if necessary

File: api.md (hunk 4 above)
Comment (dpanayotov) [reply api.md]: For consistency with Platform Registration API, the credentials are returned in the response.

File: api.md (hunk 5 above)
Comment (dpanayotov) [reply api.md]: If no update has been performed, the created_at and updated_at times will be the same.

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (dotchev) [reply README.md]: "Cloud Landscape" is referenced only once - in the Motivation section. SM is not aware of Cloud Landscapes so there is no need to define it as a term here.

**Pull request merged**